### PR TITLE
[gardening] Clean up do catch blocks

### PR DESCRIPTION
**Pull request by otaviolima**
- Remove wildcard pattern matching
- Refactor catches with empty body
- Prefer using try? to ignore errors instead of empty catch body
- Prefer not assigning a variable name

### CODE REVIEW

File: TestFoundation/TestCodable.swift [hunk 1]
```diff
@@ -404,8 +404,8 @@ class TestCodable : XCTestCase {
         for calendar in calendarValues {
             do {
                 try expectRoundTripEqualityThroughJSON(for: calendar)
-            } catch let error {
                 XCTFail("\(error) for \(calendar)")
+            } catch {
```
Comment (spevans) [inline TestFoundation/TestCodable.swift]: Should the `catch` not be 2 lines up?

Comment (otaviolima) [reply TestFoundation/TestCodable.swift]: Yes, that is correct. I messed up with another changed I made while I was picking the changes to `stage` before commit. Nice catch :)

Comment (otaviolima) [reply TestFoundation/TestCodable.swift]: Should be fine now.

**Comment by spevans** [comment]
@swift-ci please test

File: Foundation/URLSession/http/HTTPURLProtocol.swift [hunk 2]
```diff
@@ -93,7 +93,7 @@ internal class _HTTPURLProtocol: _NativeProtocol {
             case (_, nil):
                 set(requestBodyLength: .unknown)
             }
-        } catch let e {
+        } catch {
```
Comment (spevans) [inline Foundation/URLSession/http/HTTPURLProtocol.swift]: `e` is used inside the catch block for  the `errorCode(fileSystemError: e)` call so this should be left in

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
It did, but there was another error: https://ci.swift.org/view/Pull%20Request/job/swift-corelibs-foundation-PR-Linux/1153/consoleFull

File: TestFoundation/TestFileManager.swift [hunk 3]
```diff
@@ -32,21 +32,17 @@ class TestFileManager : XCTestCase {
...
         
-        ignoreError { try fm.removeItem(atPath: path) }
+        try? fm.removeItem(atPath: path)
         
         do {
             try fm.createDirectory(atPath: path, withIntermediateDirectories: false, attributes: nil)
-        } catch _ {
             XCTFail()
+        } catch {
         }
```
Comment (spevans) [inline TestFoundation/TestFileManager.swift]: Looks to be another issue with the `catch` being moved down

**Comment by otaviolima** [comment]
Fixed and ran tests locally, it is passing in my env. Sorry for the hassle.

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
Looks like it passed ok: https://ci.swift.org/view/Pull%20Request/job/swift-corelibs-foundation-PR-Linux/1154/console

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**